### PR TITLE
Renovation: editor refactoring

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

**Review by DokaRus — APPROVED** [review]

File: js/ui/editor/editor.js [hunk 1]
```diff
@@ -1,69 +1,44 @@
...
     },
 
-    _init: function() {
+    _init() {
         this.callBase();
-        this._options.cache('validationTooltipOptions', this.option('validationTooltipOptions'));
+
         const $element = this.$element();
-        $element.addClass(DX_INVALID_BADGE_CLASS);
+        const { validationTooltipOptions } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:34]:
```suggestion
        const validationTooltipOptions = this.option('validationTooltipOptions');
```

File: js/ui/editor/editor.js [hunk 2]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
             validationTooltipOptions: {}
         });
     },
 
-    _attachKeyboardEvents: function() {
-        if(!this.option('readOnly')) {
-            this.callBase();
-        }
+    _attachKeyboardEvents() {
+        const { readOnly } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:65]:
```suggestion
        const readOnly = this.option('readOnly');
```

File: js/ui/editor/editor.js [hunk 3]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
     },
 
-    _resumeValueChangeAction: function() {
+    _resumeValueChangeAction() {
         this._valueChangeActionSuppressed = false;
     },
 
-    _initMarkup: function() {
+    _initMarkup() {
+        const { name } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:91]:
```suggestion
        const name = this.option('name');
```

File: js/ui/editor/editor.js [hunk 4]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
 
-    _saveValueChangeEvent: function(e) {
+    _saveValueChangeEvent(e) {
         this._valueChangeEventInstance = e;
     },
 
-    _focusInHandler: function(e) {
-        const isValidationMessageShownOnFocus = this.option('validationMessageMode') === 'auto';
+    _focusInHandler(e) {
+        const { validationMessageMode } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:118]:
```suggestion
        const validationMessageMode = this.option('validationMessageMode ');
```

File: js/ui/editor/editor.js [hunk 5]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
+            _checkParentVisibility: false
+        }, this._options.cache('validationTooltipOptions')));
+    },
+
+    _renderValidationState() {
         const $element = this.$element();
-        let validationErrors = this.option('validationErrors');
-        if(!validationErrors && this.option('validationError')) {
-            validationErrors = [this.option('validationError')];
+        const { validationStatus, validationError, validationMessageMode } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:163]: this.option('validationStatus'), this.option('validationError'), this.option('validationMessageMode'), this.option('validationErrors'), this.option('isValid')

File: js/ui/editor/editor.js [hunk 6]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
 
-    _getValidationMessageTarget: function() {
+    _getValidationMessageTarget() {
         return this.$element();
     },
 
-    _getValidationMessagePosition: function(positionRequest) {
-        const rtlEnabled = this.option('rtlEnabled');
+    _getValidationMessagePosition(positionRequest) {
+        const { rtlEnabled, validationMessageOffset, validationBoundary } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:223]: this.option('rtlEnabled'), this.option('validationMessageOffset'), this.option('validationBoundary')

File: js/ui/editor/editor.js [hunk 7]
```diff
@@ -72,269 +47,247 @@ const Editor = Widget.inherit({
...
             my: messagePositionSide + verticalPositions[0],
             at: messagePositionSide + verticalPositions[1],
             collision: 'none flip'
         };
     },
 
-    _toggleReadOnlyState: function() {
-        const readOnly = this.option('readOnly');
+    _toggleReadOnlyState() {
+        const { readOnly } = this.option();
```
Comment (San4es) [inline js/ui/editor/editor.js:240]:
```suggestion
        const readOnly = this.option('readOnly');
```

**Pull request closed without merging**